Metalama.Framework.Engine/Services/CompilationContext.cs

```diff
@@ -136,8 +136,11 @@ internal SyntaxGenerationContext GetSyntaxGenerationContext( bool isPartial )
     /// </summary>
     public static void SetTriviaHandling( Compilation compilation, bool normalizeWhitespace, bool preserveTrivia )
     {
-        var tuple = (normalizeWhitespace, preserveTrivia);
-        _triviaHandlingDictionary.AddOrUpdate( compilation.AssemblyName.AssertNotNull(), tuple, ( _, _ ) => tuple );
+        // If trivia handling was previously set, don't reset it in case the previous computation is still running.
+        _triviaHandlingDictionary.AddOrUpdate(
```

Comment:
I don't think this solves the problem of non-determinism. First, [Memo] should be used for the whole tuple not for individual items. Second, this will in time pollute all assembly names, and all pipelines in the process will be normalizing/preserving.

Is the sharing of state between all contexts based on assembly name needed for anything? 

I think we just need a service, push it into a service provider, and query that. I didn't notice any place where this would be hard.